spec/models/form_spec.rb

```diff
@@ -32,6 +32,97 @@
     end
   end
 
+  describe "next_incomplete_section_redirect_path" do
+    let(:case_log) { FactoryBot.build(:case_log, :in_progress) }
+    let(:subsection) { form.get_subsection("household_characteristics") }
+    let(:later_subsection) { form.get_subsection("local_authority") }
+
+    context "when a user is on the check answers page for a subsection" do
+      def answer_household_needs(case_log)
+        case_log.armedforces = "No"
+        case_log.illness = "No"
+        case_log.housingneeds_a = "Yes"
+        case_log.la = "York"
+        case_log.illness_type_1 = "Yes"
+      end
+
+      def answer_tenancy_information(case_log)
+        case_log.tenancy_code = "1234"
+      end
+
+      def answer_property_information(case_log)
+        case_log.wchair = "No"
+      end
+
+      def answer_conditional_question(case_log)
+        case_log.preg_occ = "No"
+        case_log.cbl = "No"
+      end
+
+      def answer_income_and_benefits(case_log)
+        case_log.earnings = 30_000
+        case_log.incfreq = "Yearly"
+        case_log.benefits = "Some"
+        case_log.hb = "Tenant prefers not to say"
+      end
+
+      def answer_rent_and_charges(case_log)
+        case_log.period = "Every 2 weeks"
+        case_log.brent = 650
+        case_log.scharge = 0
+        case_log.pscharge = 0
+        case_log.supcharg = 0
+        case_log.tcharge = 650
+      end
+
+      def answer_local_authority(case_log)
+        case_log.layear = "1 to 2 years"
+        case_log.lawaitlist = "Less than 1 year"
+        case_log.property_postcode = "NW1 5TY"
+        case_log.reason = "Permanently decanted from another property owned by this landlord"
+        case_log.previous_postcode = "SE2 6RT"
+        case_log.mrcdate = Time.zone.parse("03/11/2019")
+      end
+
+      before do
+        case_log.tenant_code = "123"
+        case_log.age1 = 35
+        case_log.sex1 = "Male"
+        case_log.other_hhmemb = 0
+      end
+
+      it "returns the first page of the next incomplete subsection" do
```

Comment:
Is this the correct behaviour even if the first page's questions have been answered already?

Reply:
Answer to this is yes

Reply:
@paulrobertlloyd can you confirm that this is what we want rather than redirecting to the first unanswered page?

Reply:
If a section has been partially completed, I imagine we’d take you to the ‘Check your answers’ page for that section. otherwise, the first question in that section? Is that possible?

Reply:
(For the record, not a big issue either way, we can polish as we go).

Reply:
Summarising expected behaviour from discussion here:

**Scenario A. User is on Section 2 Check Answers page:**

Section 1: Not started
Section 2: Completed
Section 3: In Progress

-> Link should take the user to Section 3 Check answers page

**Scenario B. User is on Section 2 Check Answers page:**

Section 1: Not started
Section 2: Completed
Section 3: Not started

-> Link should take the user to Section 3 first question page

**Scenario C. User is on Section 2 Check Answers page:**

Section 1: Not started
Section 2: Completed
Section 3: Completed

-> Link should take the user to Section 1 first question page